src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,90 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
+@SuppressWarnings("unchecked")
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private Node<K, V>[] table;
+    private int threshold;
+    private int capacity;
+    private int size;
+
+    public MyHashMap() {
+        this.table = new Node[DEFAULT_CAPACITY];
```

Comment:
[where do we need `this`?](https://stackoverflow.com/a/2411283)

Reply:
sorry, I read the article, but I thought that if we have to use "this" in setters it also means that we should use them in constructors to get rid of the ambiguity